packages/react-dom/src/__tests__/ReactMultiChildReconcile-test.js

```diff
@@ -221,24 +222,40 @@ function verifyDomOrderingAccurate(outerContainer, statusDisplays) {
   expect(orderedDomKeys).toEqual(orderedLogicalKeys);
 }
 
-function testPropsSequenceWithPreparedChildren(sequence, prepareChildren) {
+async function testPropsSequenceWithPreparedChildren(
+  sequence,
+  prepareChildren,
+) {
   const container = document.createElement('div');
-  const parentInstance = ReactDOM.render(
-    <FriendsStatusDisplay {...sequence[0]} prepareChildren={prepareChildren} />,
-    container,
-  );
+  const root = ReactDOMClient.createRoot(container);
+  let parentInstance;
+  await act(() => {
+    root.render(
+      <FriendsStatusDisplay
+        {...sequence[0]}
+        prepareChildren={prepareChildren}
+        ref={current => {
+          if (parentInstance === undefined) {
+            parentInstance = current;
```

Comment:
Nit: can this assignment be unconditional?

Reply:
Either I make this conditional or I have to repeat the `ref` for each render call.

In other tests we sometimes went with assigning ref in cDM instead to not have to repeat the ref prop but that isn't possible here since `FriendsStatusDisplay` is out of scope.


The current approach is closest to the original test and similar to the oter cDM patterns so I'm leaving it.